packages/autorest.python/autorest/codegen/models/model_type.py

```diff
@@ -79,18 +82,33 @@ def serialization_type(self) -> str:
             )
         return "object"
 
+    @property
+    def _is_polymorphic(self) -> bool:
+        return any(p.is_polymorphic for p in self.properties)
+
     def type_annotation(self, **kwargs: Any) -> str:
         if self.code_model.options["models_mode"]:
             is_operation_file = kwargs.pop("is_operation_file", False)
             retval = f"_models.{self.name}"
             if not self.is_public:
                 retval = f"{self.code_model.models_filename}.{retval}"
+            if self.code_model.options["models_mode"] == "dpg" and kwargs.pop(
+                "is_body_parameter", False
+            ):
+                retval += ", JSON"
+                if not kwargs.pop("in_combined_type", False):
+                    retval = f"Union[{retval}]"
             return retval if is_operation_file else f'"{retval}"'
         return "ET.Element" if self.is_xml else "JSON"
 
     def docstring_type(self, **kwargs: Any) -> str:
         if self.code_model.options["models_mode"]:
-            return f"~{self.code_model.namespace}.models.{self.name}"
+            ret = f"~{self.code_model.namespace}.models.{self.name}"
+            if self.code_model.options["models_mode"] == "dpg" and kwargs.pop(
+                "is_body_parameter", False
```

Comment:
we should add the `" or JSON"` information in the `docstring_type` for body parameter

Reply:
Hi @iscai-msft , shall we have some discuss on this:

When the bodyparameter is a CombinedType, I think we need it to be `InputModel or JSON or IO` instead of `InputModel or IO or JSON`?

It seems be hard to achieve this if we add `or JSON` in BodyParameter, because we will need to dissemble the unioned doc string `InputModel or IO` and insert ` or JSON` at the right place. And doing this seems make doc string logics couple together tightly accross CombinedType/ModelType/BodyParameter.